### PR TITLE
add performance section to dev guide

### LINKED ISSUE #9919: Develop a performance strategy document describing our preferred approaches and tools
At a recent tech hours we started discussing some strategies for performance monitoring during the dev cycle.

This issue is to capture our desired strategies in a document that we can review and revise for future use. The person picking this up would start the document and, of course, reach out, to others to comment and edit, as needed. We can then review again until we have the final document.

### PR DESCRIPTION
**Pull request by pdurbin**
**What this PR does / why we need it**:

As explained in the issue, we'd like to capture our desired strategies in a document that we can review and revise for future use.

**Which issue(s) this PR closes**:

- Closes #9919

**Special notes for your reviewer**:

None. We went over [the original doc](https://docs.google.com/document/d/1Lbl0NJaoV6WOYmP-Mw6UZaA9KYSyliu2ABgs2xNdOKU/edit?usp=sharing) at tech hours. This is mostly formatting it at rst.

**Suggestions on how to test this**:

Make sure the docs build.

**Does this PR introduce a user interface change? If mockups are available, please link/include them here**:

No.

**Is there a release notes update needed for this change?**:

Hmm, I don't think so. Maybe an email to the mailing list.

**Additional documentation**:

It is documentation.

### CODE REVIEW

File: doc/sphinx-guides/source/developers/performance.rst [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+
+Command Engine Execution Rate Metering
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We'd like to rate limit commands (CreateDataset, etc.) so that we can keep them at a reasonable level (`#9356 <https://github.com/IQSS/dataverse/issues/9356>`_). This is similar to how many APIs are rate limited, such as the GitHub API.
+
+Solr
+~~~~
+
+While in the past Solr performance hasn't been much of a concern, in recent years we've noticed performance problems when Harvard Dataverse is under load. We are investigating in `#9635 <https://github.com/IQSS/dataverse/issues/9635>`_.
```
Comment (sekmiller) [inline doc/sphinx-guides/source/developers/performance.rst:119]: This issue is marked Closed as Completed

File: doc/sphinx-guides/source/developers/performance.rst [hunk 2]
```diff
@@ -0,0 +1,196 @@
...
+
+Learn from Training and Conferences
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Most likely there is training available that is oriented toward performance. The subject of performance often comes up at conferences as well.
+
+Learn from the Community How They Monitor Performance
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Some members of the Dataverse community are likely users of newish tools like the ELK stack (Elasticsearch, Logstash, and Kibana), the TICK stack (Telegraph InfluxDB Chronograph and Kapacitor), GoAccess, Prometheus, Graphite, and more we haven't even heard of. In the :doc:`/admin/monitoring` section of the Admin Guide, we already encourage the community to share findings (, but we could dedicate time to this topic at our annual meeting or community calls.
```
Comment (sekmiller) [inline doc/sphinx-guides/source/developers/performance.rst:186]: Close parens here?

**Review by sekmiller — COMMENTED** [review]
Looks Ok. A couple of the issues referred to are already closed as completed

File: doc/sphinx-guides/source/developers/performance.rst (hunk 1 above)
Comment (pdurbin) [reply doc/sphinx-guides/source/developers/performance.rst]: Thanks in 80634c7 I linked to the PR.

File: doc/sphinx-guides/source/developers/performance.rst [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+Solr
+~~~~
+
+While in the past Solr performance hasn't been much of a concern, in recent years we've noticed performance problems when Harvard Dataverse is under load. We are investigating in `#9635 <https://github.com/IQSS/dataverse/issues/9635>`_.
+
+Datasets with Large Numbers of Files or Versions
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We'd like to scale Dataverse to better handle large number of files or versions (`#9763 <https://github.com/IQSS/dataverse/issues/9763>`_).
+
```
[the remark this answers is not in the document]
Comment (pdurbin) [reply doc/sphinx-guides/source/developers/performance.rst:125]: Good catch, in 80634c7 I linked to the PR as an example.

File: doc/sphinx-guides/source/developers/performance.rst (hunk 2 above)
Comment (pdurbin) [reply doc/sphinx-guides/source/developers/performance.rst]: Whoops. I removed the stray ( in 80634c7. Thanks.

**Review by sekmiller — APPROVED** [review]
Thanks for the updates

**Pull request merged**